rtribbia/main.js

```diff
@@ -0,0 +1,76 @@
+// Your card and deque factory code should be loaded into the browser before this file
+// by the main.html container file.
+// At this point, both makeCard and makeDeque should be defined.
+
+
+// 2b:
+// make a deque instance to store a full deck of cards:
+var deckOfCards = makeDeque(makeCard.fullSet);
+
+function dispnames () { deckOfCards.values.forEach(function (x) {console.log(x.name());}); }
+
+function suitSort(a,b) {
+	var a_s = a.suit();
+	var b_s = b.suit();
+	var a_r = a.rank();
+	var b_r = b.rank();
+	if ((a_s > b_s) || ((a_r > b_r) && (a_s == b_s))) { //a & b are in order if: A.suit > b.suit  OR (A.rank > B.rank AND A.suit equals B.suit)
+		return 1;
+	} else if ((b_s > a_s) || ((b_r > a_r) && (a_s == b_s))) { //a & b are OUT of order if: B.suit > A.suit  OR (B.rank > A.rank AND A.suit equals S.suit)
+		return -1;
+	} else {
+		return 0;
+	}
+}
+
+function alphaSort(a,b) {
```

Comment:
First, I'd strongly suggest using more human-readable variable names&mdash;even something like `a_suit` and `b_rank` would be more comprehensible.

Second, this is only checking the first letter of rank or suit, meaning that all the "six"es and "seven"s are lumped together, as are all the "two"s, "three"s, and "ten"s. You'd be better off comparing whole words rather than just individual characters.